plugin/src/main/java/org/opensearch/ml/model/MLModelAutoReLoader.java

```diff
@@ -0,0 +1,285 @@
+                ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+                MLTask mlTask = MLTask.parse(parser);
+
+                String workerNode = mlTask.getWorkerNode();
+                MLTaskType mlTaskType = mlTask.getTaskType();
+                MLTaskState mlTaskState = mlTask.getState();
+
+                if (workerNode.contains(nodeId) && mlTaskType == MLTaskType.LOAD_MODEL && mlTaskState == MLTaskState.COMPLETED) {
+                    autoReLoadModelByNodeAndModelId(nodeId, mlTask.getModelId());
+                }
+            } catch (RuntimeException | IOException e) {
+                reTryTimes++;
+                // Store the latest value of the reTryTimes and node id under the index ".plugins-ml-model-reload"
+                saveLatestReTryTimes(nodeId, reTryTimes);
+                log.error("Can't auto reload model in node id {} ,has try {} times\nThe reason is:{}", nodeId, reTryTimes, e.getMessage());
+            }
+        }
+
+        // Store the latest value of the reTryTimes and node id under the index ".plugins-ml-model-reload"
+        saveLatestReTryTimes(nodeId, reTryTimes);
+    }
+
+    /**
+     *  auto reload 1 model under the node id
+     * @param nodeId node id
+     * @param modelId model id
+     */
+    @VisibleForTesting
+    void autoReLoadModelByNodeAndModelId(String nodeId, String modelId) {
+        MLLoadModelRequest mlLoadModelRequest = new MLLoadModelRequest(modelId, new String[] { nodeId }, false, false);
+        client.execute(MLLoadModelAction.INSTANCE, mlLoadModelRequest).actionGet();
+    }
+
+    /**
+     * get retry times from the index ".plugins-ml-model-reload" by 1 ml node
+     * @param nodeId the filter condition to query
+     * @return retry times
+     */
+    @VisibleForTesting
+    Integer getReTryTimes(String nodeId) {
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+        searchSourceBuilder.fetchSource(new String[] { MODEL_LOAD_RETRY_TIMES_FIELD }, null);
+        QueryBuilder queryBuilder = QueryBuilders.termQuery("_id", nodeId);
+        searchSourceBuilder.query(queryBuilder);
+        SearchRequest searchRequest = new SearchRequest().source(searchSourceBuilder).indices(ML_MODEL_RELOAD_INDEX);
+
+        try {
+            SearchResponse response = client.execute(SearchAction.INSTANCE, searchRequest).actionGet(5000);
+
+            if (response == null) {
+                throw new RuntimeException("can't get retry times by " + nodeId);
+            }
+            SearchHits searchHits = response.getHits();
+            if (searchHits == null) {
+                throw new RuntimeException("can't get retry times by " + nodeId);
+            }
+            SearchHit[] hits = searchHits.getHits();
+            if (CollectionUtils.isEmpty(hits)) {
+                return 0;
+            }
+
+            for (SearchHit hit : hits) {
```

Comment:
Check the hits size and if not expected then throw exception rather than throw at the bottom of the method.

Reply:
because I will use `idsQuery`, so the result will return a distinct result, the size of hits will be 0 or 1,and before this code,I check the situation of 0,so in this code,the size of hits must be 1

Reply:
Add assertion here or using `hits.get(0)` instead of for loop and leaking condition to the end of the method.

Reply:
cool, it is ok. I will changed it.